File: apps/go-idp/internal/auth/argon2.go
package auth

import (
	"errors"

	"golang.org/x/crypto/bcrypt"
)

const bcryptCost = 10

func HashPassword(plain string) (string, error) {
	h, err := bcrypt.GenerateFromPassword([]byte(plain), bcryptCost)
	if err != nil {
		return "", err
	}
	return string(h), nil
}

func VerifyPassword(encoded, plain string) (bool, error) {
	err := bcrypt.CompareHashAndPassword([]byte(encoded), []byte(plain))
	if err == nil {
		return true, nil
	}
	if errors.Is(err, bcrypt.ErrMismatchedHashAndPassword) {
		return false, nil
	}
	return false, err
}

File: apps/node-idp/src/auth/argon2.ts
import bcrypt from 'bcrypt';

const ROUNDS = 10;

export function hashPassword(plain: string): Promise<string> {
  return bcrypt.hash(plain, ROUNDS);
}

export function verifyPassword(hashed: string, plain: string): Promise<boolean> {
  return bcrypt.compare(plain, hashed);
}